pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -106,13 +125,19 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
+	if string(instance.Status.Phase) == "" {
+		r.updatePhase(instance, hcov1alpha1.InstallingPhase, request, "")
+	} else {
+		r.updatePhase(instance, hcov1alpha1.ReconcilingPhase, request, "")
+	}
 
 	// Define KubeVirt's configuration ConfigMap first
 	kvConfig := newKubeVirtConfigForCR(instance)
 	kvConfig.ObjectMeta.Namespace = request.Namespace
 
 	// Set HyperConverged instance as the owner and controller
 	if err := controllerutil.SetControllerReference(instance, kvConfig, r.scheme); err != nil {
+		r.updatePhase(instance, hcov1alpha1.FailedPhase, request, "Set HyperConverged instance as the owner and controller")
```

Comment:
Instead of having to add phase on each err, we could create a shim in front of the `Reconcile` func which captures the return from `Reconcile` and processes it.